packages/js-client/test/unit/context.test.ts

```diff
@@ -0,0 +1,291 @@
+import { Wallet } from '@ethersproject/wallet';
+import { GraphQLClient } from 'graphql-request';
+
+// @ts-ignore
+declare const describe, it, beforeEach, expect;
+
+describe('Context instances', () => {
+  let contextParams: SimpleStorageContextParams;
+  const TEST_WALLET =
+    '8d7d56a9efa4158d232edbeaae601021eb3477ad77b5f3c720601fd74e8e04bb';
+  const web3endpoints = {
+    working: ['https://cloudflare-eth.com/'],
+    failing: ['https://bad-url-gateway.io/'],
+  };
+  beforeEach(() => {
+    // reset contextParams
+    contextParams = {
+      network: 'mainnet',
+      signer: new Wallet(TEST_WALLET),
+      daoFactoryAddress: '0x1234',
+      web3Providers: web3endpoints.working,
+      gasFeeEstimationFactor: 0.1,
+      graphqlNodes: [],
+      ipfsNodes: [],
+      myParam: 'notDefault',
+    };
+  });
+  it('Should create an empty context and have default values', () => {
+    const context = new SimpleStorageContext();
+    expect(context).toBeInstanceOf(SimpleStorageContext);
+    expect(context.network.name).toBe('homestead');
+    expect(context.network.chainId).toBe(1);
+    expect(context.daoFactoryAddress).toBe(LIVE_CONTRACTS.homestead.daoFactory);
+    expect(context.ensRegistryAddress).toBe(context.network.ensAddress);
+    expect(context.gasFeeEstimationFactor).toBe(0.625);
+    expect(context.web3Providers.length).toBe(0);
+    expect(context.ipfs.length).toBe(IPFS_NODES.homestead.length);
+    expect(context.graphql.length).toBe(GRAPHQL_NODES.homestead.length);
+    expect(context.myParam).toBe('default');
+    context.web3Providers.map((provider) => {
+      expect(provider).toBeInstanceOf(JsonRpcProvider);
+    });
+    context.ipfs.map((ipfsClient) => {
+      expect(ipfsClient).toBeInstanceOf(IpfsClient);
+    });
+    context.graphql.map((graphqlClient) =>
+      expect(graphqlClient).toBeInstanceOf(GraphQLClient)
+    );
+  });
+  it('Should create a context and have the correct values', () => {
+    const context = new SimpleStorageContext(contextParams);
+
+    expect(context).toBeInstanceOf(SimpleStorageContext);
+    expect(context.network.name).toBe('homestead');
+    expect(context.network.chainId).toBe(1);
+    expect(context.daoFactoryAddress).toBe(contextParams.daoFactoryAddress);
+    expect(context.ensRegistryAddress).toBe(context.network.ensAddress);
+    expect(context.gasFeeEstimationFactor).toBe(
+      contextParams.gasFeeEstimationFactor
+    );
+    context.web3Providers.map((provider) =>
+      expect(provider).toBeInstanceOf(JsonRpcProvider)
+    );
+    context.ipfs.map((ipfsClient) =>
+      expect(ipfsClient).toBeInstanceOf(IpfsClient)
+    );
+    context.graphql.map((graphqlClient) =>
+      expect(graphqlClient).toBeInstanceOf(GraphQLClient)
+    );
+    expect(context.myParam).toBe('notDefault');
+  });
+  it('Should set a new context and have the correct values', () => {
+    const context = new SimpleStorageContext(contextParams);
+    contextParams = {
+      network: 'goerli',
```

Comment:
SupportedNetworks.GOERLI ?

Reply:
👍

Reply:
Yes, but we support Inputing a `Networkish` then we do a `getNetwork` to transform it to our array, else users would need to pass `homestrad` instead of `mainnet` and we wouldn't supoort passing the chain id

Reply:
I see
feel free to resolve then